app/views/users/new.html.haml

```diff
@@ -2,10 +2,14 @@
   .row
     .col-sm-10.col-sm-offset-1
       = bootstrap_form_for @user, layout: :horizontal, label_col: 'col-sm-2', control_col: 'col-sm-6' do |f|
+        = f.hidden_field :invite_token, value: @invitation.token if @invitation
         %header
           %h1 Register
         %fieldset
-          = f.email_field :email, label: 'Email Address'
+          - if @invitation
+            = f.email_field :email, label: 'Email Address', value: @invitation.email
+          - else
+            = f.email_field :email, label: 'Email Address'
```

Comment:
This could rather be one liner.

`= f.email_field :email, label: 'Email Address', value: @invitation&.email` if you are using ruby 2.3.0+

or 

`= f.email_field :email, label: 'Email Address', value: (@invitation.email if @invitation)`